docs/PROCEDURES.md

```diff
@@ -0,0 +1,69 @@
+# Additional procedures for code maintenance 
+
+## Managing pinned dependencies
+
+The OpenSSF, via the [scorecard](https://securityscorecards.dev/) project recommends that projects pin any
+dependencies they use:
+
+* to ensure reproducibility
+* to reduce the change for rogue dependency updates to compromise software
+
+It's important to note that this requires any changes to dependencies are properly reviews, and
+these changes, by design, should not be automatic in themselves, though automated tools may provide recommendations.
+
+### Python dependencies
+
+Python dependencies used in the build process should be pinned to a specific version to ensure reproducibility.
+
+This is done by:
+ * Using the `--require-hashes` option on any `pip install` command line
+ * Adding the required hash in the `requirements.txt`
+
+ Currently this is used withou `.github/workflows` but the same principle applies elsewhere.
+
+ To make this easier, a version of the `requirements.txt` without hashes has been saved as `requirements.in`. This is
+ to make maintenance easier, but it is not used at script execution time.
+
+ The `pip-compile` tool must be installed via the [pip-tools](https://pypi.org/project/pip-tools/) package.
+ 
+ To add a new, or changed dependency:
+ ```
+ pip-compile --generate-hashes --output-file=requirements.txt requirements.in
+ ```
+
+ This will update requirements.txt with the correct hashes.
+
+ Correction function should be verified, and then both `requirements.txt` and `requirements.in` checked in.
```

Comment:
I do not understand this sentence: What is a "Correction function" and what does one (contributor and/or reviewer) have to do to verify it?

Reply:
A typo. It should say 'correct functionality'